crates/net/p2p/src/test_utils/full_block.rs

```diff
@@ -97,10 +97,14 @@ impl HeadersClient for NoopFullBlockClient {
 /// This full block client can be [Clone]d and shared between multiple tasks.
 #[derive(Clone, Debug)]
 pub struct TestFullBlockClient {
-    headers: Arc<Mutex<HashMap<B256, Header>>>,
-    bodies: Arc<Mutex<HashMap<B256, BlockBody>>>,
-    // soft response limit, max number of bodies to respond with
-    soft_limit: usize,
+    /// Headers in memory
+    pub headers: Arc<Mutex<HashMap<B256, Header>>>,
+    /// Bodies in memory
+    pub bodies: Arc<Mutex<HashMap<B256, BlockBody>>>,
+    /// Soft response limit, max number of bodies to respond with
+    pub soft_limit: usize,
+    /// Whether return bad block body
+    pub bad_block_body: bool,
```

Comment:
this isn't necessary.
we can test this by inserting bad blocks manually if we add an insert function that also accepts a hash